doc/crypto/api.db/psa/crypto.h

```diff
@@ -17,6 +17,17 @@ typedef uint8_t psa_key_persistence_t;
 typedef uint16_t psa_key_type_t;
 typedef uint32_t psa_key_usage_t;
 typedef /* implementation-defined type */ psa_mac_operation_t;
+typedef /* implementation-defined type */ psa_pake_cipher_suite_t;
+typedef uint8_t psa_pake_family_t;
+typedef /* implementation-defined type */ psa_pake_operation_t;
+typedef uint32_t psa_pake_primitive_t;
+typedef uint8_t psa_pake_primitive_type_t;
+typedef uint8_t psa_pake_role_t;
+typedef uint8_t psa_pake_step_t;
+typedef struct psa_key_production_parameters_t {
+    uint32_t flags;
+    uint8_t data[];
```

Comment:
Yesterday I learned that [flexible array members are not standard C++](https://github.com/Mbed-TLS/mbedtls/issues/9020). It is generally expected that C++ compilers can consume C headers, so Mbed TLS and PSA should avoid making headers incompatible with C++.

In Mbed TLS, we'll stay with this API in 3.6.x LTS (too late, it's released), but we'd like to change the API to be C++-compatible in our next major release.

The obvious fix would be to separate the variable-length data out of the struct. This may require a little more RAM in client-server scenarios, but on the bright side it would save the hassle around mixing fixed-format and variable-length data.

Reply:
> The obvious fix would be to separate the variable-length data out of the struct. This may require a little more RAM

The most obvious is declaring array of size 1 instead of 0; RAM waste would be none or minimal.
Static analysis and memory sanitizer might be unhappy, though.